tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/mapper/ProgrammeMembershipMapper.java

```diff
@@ -34,166 +32,187 @@ public class ProgrammeMembershipMapper {
-    result.setLeavingReason(programmeMembershipDto.getLeavingReason());
-    result.setTrainingPathway(programmeMembershipDto.getTrainingPathway());
-    result.setAmendedDate(programmeMembershipDto.getAmendedDate());
     Programme programme = null;
-    if (programmeMembershipDto.getProgrammeId() != null) {
+    if (dto.getProgrammeId() != null) {
       programme = new Programme();
-      programme.setId(programmeMembershipDto.getProgrammeId());
-      programme.setProgrammeName(programmeMembershipDto.getProgrammeName());
-      programme.setOwner(programmeMembershipDto.getProgrammeOwner());
-      programme.setProgrammeNumber(programmeMembershipDto.getProgrammeNumber());
-      result.setRotation(rotationDTOToRotation(programmeMembershipDto.getRotation()));
+      programme.setId(dto.getProgrammeId());
+      programme.setProgrammeName(dto.getProgrammeName());
+      programme.setOwner(dto.getProgrammeOwner());
+      programme.setProgrammeNumber(dto.getProgrammeNumber());
+      entity.setRotation(rotationMapper.toEntity(dto.getRotation()));
     }
-    result.setProgramme(programme);
-    result.setTrainingNumber(
-        trainingNumberDTOToTrainingNumber(programmeMembershipDto.getTrainingNumber()));
-    result.setPerson(personDTOToPerson(programmeMembershipDto.getPerson()));
+    entity.setProgramme(programme);
 
-    if (programmeMembershipDto.getCurriculumMemberships() == null) {
-      result.setCurriculumMemberships(new HashSet<>());
-    }
-    result.getCurriculumMemberships()
-        .addAll(curriculumMembershipMapper.toEntity(programmeMembershipDto));
-    result.getCurriculumMemberships().forEach(cm -> cm.setProgrammeMembership(result));
-    return result;
+    entity.setTrainingNumber(
+        trainingNumberMapper.trainingNumberDTOToTrainingNumber(dto.getTrainingNumber()));
+    entity.setPerson(personDTOToPerson(dto.getPerson()));
+
+    entity.setCurriculumMemberships(new HashSet<>());
+    entity.getCurriculumMemberships()
+        .addAll(curriculumMembershipMapper.toEntity(dto));
+    entity.getCurriculumMemberships().forEach(cm -> cm.setProgrammeMembership(entity));
+    return entity;
   }
 
+  /**
+   * Convert ProgrammeMembership DTOs to entities by using {@link #toEntity(ProgrammeMembershipDTO)}
+   * for each DTO.
+   *
+   * @param pmDtos list of ProgrammeMembershipDTO
+   * @return the ProgrammeMembership entity list
+   */
   public List<ProgrammeMembership> programmeMembershipDTOsToProgrammeMemberships(
-      List<ProgrammeMembershipDTO> programmeMembershipDTOs) {
-    List<ProgrammeMembership> result = Lists.newArrayList();
-
-    for (ProgrammeMembershipDTO programmeMembershipDto : programmeMembershipDTOs) {
-      result.add(toEntity(programmeMembershipDto));
-    }
-
-    return result;
+      List<ProgrammeMembershipDTO> pmDtos) {
+    return pmDtos.stream().map(this::toEntity).collect(Collectors.toList());
   }
 
+  // convert programmeMembership entity to DTO with an empty curriculumMembership list
```

Comment:
It would be good to make clear if/why the curriculum memberships are not added to the programme membership here.  I would have assumed it was a bit of legacy that we didn't want to continue but don't know.

Let me know if I should look into why we'd want to convert the programme membership without the nested curriculum memberships.

Reply:
I thought methods should be annotated with full "JavaDoc" comments too!